workflow/controller/workflowpod.go

```diff
@@ -1112,6 +1129,20 @@ func createSecretVolumes(tmpl *wfv1.Template) ([]apiv1.Volume, []apiv1.VolumeMou
 		createSecretVolume(allVolumesMap, art, uniqueKeyMap)
 	}
 
+	if tmpl.Data != nil {
+		spot := -1
+		for i, step := range tmpl.Data {
+			if step.WithArtifactPaths != nil {
+				spot = i
+				break
+			}
+		}
+		if spot >= 0 {
+			log.Infof("SIMON creating volume")
+			createSecretVolume(allVolumesMap, tmpl.Data[spot].WithArtifactPaths.Artifact, uniqueKeyMap)
```

Comment:
for PodTemplate I achieved something similar by adding `Template.GetVolumes()` - I wonder if that would work here too?

Reply:
This was a rough and ready implementation. Was not meant to be final. Was going to re implement before making this PR ready for review